flask/app/variants.py

```diff
@@ -228,17 +227,151 @@ def parse_gene_panel() -> List[Any]:
+        app.logger.error("Invalid region format: %s", region)
+        abort(400, description="Invalid region format")
+
+    # prepare set of regions
+    region_list = []
+    for r in set([match.groups() for match in matches]):
+        # fix regions to be consistent (start-end)
+        if int(r[START]) > int(r[END]):
+            region_list.append((r[CHR], r[END], r[START]))
+        else:
+            region_list.append(r)
+
+    # query
+    region_filter = or_(
+        *[
+            and_(
+                models.Variant.chromosome == tup[CHR],
+                int(tup[START]) <= models.Variant.position,
+                models.Variant.position <= int(tup[END]),
+            )
+            for tup in region_list
+        ]
+    )
+
+    return region_filter
+
+
+def parse_position(position: str):
+    """
+    Parses a given list of positions and chromosomes (eg. "chr1:5000,chrY:8000").
+    Returns a filter for use against the variant table to find corresponding variants
+    based on gene start and end positions.
+    """
+    CHR, POS = 0, 1
+    # format check
+    # 1: chromosome, 2: position
+    position_pattern = re.compile(r"^chr([\da-zA-Z]+):([\d]+)$")
+    matches = [position_pattern.match(p) for p in position.split(",")]
+
+    app.logger.info("Requested position: %s", position)
+
+    if not all(match is not None and len(match.groups()) == 2 for match in matches):
+        app.logger.error("Invalid position format: %s", position)
+        abort(400, description="Invalid position format")
+
+    # prepare set of positions
+    position_set = set([match.groups() for match in matches])
+
+    # query
+    position_filter = or_(
+        *[
+            and_(
+                models.Variant.chromosome == tup[CHR].upper(),
+                models.Variant.position == tup[POS],
+            )
+            for tup in position_set
+        ]
+    )
+
+    return position_filter
+
+
+def parse_rsid(rsid: str):
+    rsid_pattern = re.compile(r"^rs[\d]+$")
+    matches = [rsid_pattern.match(p) for p in rsid.split(",")]
+
+    if not all(match is not None for match in matches):
+        app.logger.error("Invalid rsID format: %s", rsid)
+        abort(400, description="Invalid rsID format")
+
+    rsid_set = set(["%{}%".format(match.group()) for match in matches])
+
+    rsid_filter = or_(*[models.Variant.rsids.like(r) for r in rsid_set])
+
+    return rsid_filter
+
+
+def parse_requested_filter(search_type: str, param: str):
+    if search_type == "genes":
+        return parse_gene_panel(param)
+    if search_type == "regions":
+        return parse_region(param)
+    if search_type == "positions":
+        return parse_position(param)
+    if search_type == "rsids":
+        return parse_rsid(param)
+
+    # This shouldn't happen
+    app.logger.error("Invalid search type '%s'", search_type)
+    abort(400, description="Invalid parameter")
+
+
+def jsonify_variant_wise(tup, columns):
```

Comment:
for book keeping purposes, do you mind leaving a comment that it's unused as i don't see this being referenced, please ignore this if it's already mentioned somewhere else, eg. design docs!

Reply:
This function is left over from trying to implement column selection in JSON format. It's unused and I don't think it even works, so I'll just remove it.